### ISSUE TITLE
rule: no-require-esmodule?

### ISSUE THREAD

**Issue by benmosher** [benmosher/eslint-plugin-import#222]
Report CommonJS `require`, but only for modules.

Could be a mode of `no-commonjs`.

Given:

``` js
// ./foo.js
export const foo = "bar"

// ./x.js
module.exports  = { x: "y" }
```

then:

``` js
const foo = require('./foo') // this should be reported
    , x = require('./x') // this should not
```

**Comment by lo1tuma** [comment]
What should happen if `./foo.js`

* has both `export` and `module.exports`/`exports`?
* has no export at all?

**Comment by benmosher** [comment]
I think both would be reported, and neither would not.

No exports case is arguably covered by `no-commonjs` already.

This certainly hearkens back to the ongoing discussion in Node core about how to determine if a file should be a module. 😅

**Comment by ljharb** [comment]
Given that node is going with `.mjs`, and it's almost definitely going to work just fine to require ESM or import CJS, this seems no longer needed.

**Comment by fregante** [comment]
This could still be useful to push developers to import ES Modules correctly as they become available and make future migrations to `no-commonjs` seem less daunting (because all ES Modules will already be imported via `import`)
